permission_handler_web/lib/web_delegate.dart

```diff
@@ -0,0 +1,223 @@
+import 'dart:html' as html;
+import 'dart:async';
+
+import 'package:permission_handler_platform_interface/permission_handler_platform_interface.dart';
+
+/// The delegate class for WebPermissionHandler.
+/// Used for dependency injection of html.MediaDevices and html.Permissions objects
+class WebDelegate {
+  /// Constructs a WebDelegate.
+  WebDelegate(this.devices, this.geolocation, this.htmlPermissions);
+
+  /// The html media devices object used to request camera and microphone permissions.
+  html.MediaDevices? devices;
+
+  /// The html geolocation object used to request location permission.
+  html.Geolocation? geolocation;
+
+  /// The html permissions object used to check permission status.
+  html.Permissions? htmlPermissions;
+
+  /// Getter for WebDelegate.
+  WebDelegate get webDelegate =>
+      WebDelegate(devices, geolocation, htmlPermissions);
```

Comment:
this getter seems in the wrong place, I would expect it in the `PermissionHandlerWeb` class and be implemented as a way of lazy loading. Now it basically for works as a constructor creating new instances each time it is being called.